apps/backend/src/events/events.service.ts

```diff
@@ -11,12 +12,28 @@ export class EventsService {
   constructor(readonly prisma: PrismaService) {}
 
   async create(data: CreateEventDto, userId: string): Promise<Event> {
-    return await this.prisma.event.create({
-      data: {
-        ...data,
-        ownerId: userId,
-      },
-    });
+    const { drinkActions, ...eventData } = data;
+    try {
+      return await this.prisma.event.create({
+        data: {
+          ...eventData,
+          ownerId: userId,
+          drinkActions: {
+            create:
+              drinkActions?.map((action) => ({
+                ...action,
+                drink: { connect: { id: action.drinkId } },
+              })) || [],
+          },
+        },
+        include: { drinkActions: true },
+      });
+    } catch (error) {
+      if (error instanceof PrismaClientKnownRequestError && error.code === 'P2025') {
+        throw new BadRequestException('drinkId or eventId does not exist');
+      }
+      throw error;
```

Comment:
if you test this, you'll see that the error code is different for foreign key constraint violation.

Reply:
i tested it, and maybe i did it wrong, because i still got P2025 when i used an invalid drinkid. when i used P2003 it was internal server error:l

Reply:
You're right, P2025 is the correct code. I must've did something wrong last time